api/controllers/loomies.go

```diff
@@ -303,3 +306,126 @@ func HandleFuseLoomies(c *gin.Context) {
 		"message": "Loomies fused successfully",
 	})
 }
+
+func HandleCaptureLoomie(c *gin.Context) {
+	loomie_req := interfaces.LoomieForm{}
+	userid, _ := c.Get("userid")
+
+	if err := c.BindJSON(&loomie_req); err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": true, "message": "Bad request"})
+		return
+	}
+
+	if loomie_req.LoomieId == "" {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": true, "message": "No Loomie ID was provided"})
+		return
+	}
+
+	loomie, err := models.ValidateLoomieExists(loomie_req.LoomieId)
+
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			c.AbortWithStatusJSON(http.StatusNotFound, gin.H{"error": true, "message": "Loomie was not found"})
+			return
+		} else {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": true, "message": "Internal server error"})
+			return
+		}
+	}
+
+	isNear := utils.IsNear(interfaces.Coordinates{
+		Latitude:  loomie.Latitude,
+		Longitude: loomie.Longitude,
+	}, interfaces.Coordinates{
+		Latitude:  loomie_req.Latitude,
+		Longitude: loomie_req.Longitude,
+	})
+
+	if !isNear {
+		c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"error": true, "message": "User is not near the loomie"})
+		return
+	}
+
+	user, err := models.GetUserById(userid.(string))
+
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			c.AbortWithStatusJSON(http.StatusNotFound, gin.H{"error": true, "message": "User was not found"})
+			return
+		} else {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": true, "message": "Internal server error"})
+			return
+		}
+	}
+
+	mongoid, err := primitive.ObjectIDFromHex(loomie_req.LoomieBallId)
+
+	if err != nil {
```

Comment:
Maybe this can also be a `400` error, usually an `ObjectIDFromHex` error occurs when the string isn't a valid Mongo Id, so you can respond with an error like "The given Loomball id is not valid".